semantic-core/src/Data/Core/Parser.hs

```diff
@@ -48,65 +50,79 @@ core :: (TokenParsing m, Monad m) => m (Term Core User)
 core = expr
 
 expr :: (TokenParsing m, Monad m) => m (Term Core User)
-expr = atom `chainl1` go where
-  go = choice [ (Core....) <$ dot
-              , (Core.$$)  <$ notFollowedBy dot
-              ]
+expr = ifthenelse <|> lambda <|> rec <|> load <|> assign
+
+assign :: (TokenParsing m, Monad m) => m (Term Core User)
+assign = application <**> (flip (Core..=) <$ symbolic '=' <*> application <|> pure id) <?> "assignment"
```

Comment:
Can we pull this out into a where-clause? Tough to read as is.

Reply:
Fixed in b80dd5381.